File: robot_examples/javascript/rest_api/src/model/RobotConfiguration.js
(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['../ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.NetRobotsApi) {
      root.NetRobotsApi = {};
    }
    root.NetRobotsApi.RobotConfiguration = factory(root.NetRobotsApi.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The RobotConfiguration model module.
   * @module model/RobotConfiguration
   * @version 2.0.0
   */

  /**
   * Constructs a new <code>RobotConfiguration</code>.
   * Configurations of the robot.\n
   * @alias module:model/RobotConfiguration
   * @class
   * @param name
   */
  var exports = function(name) {

    this['name'] = name;










  };

  /**
   * Constructs a <code>RobotConfiguration</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/RobotConfiguration} obj Optional instance to populate.
   * @return {module:model/RobotConfiguration} The populated <code>RobotConfiguration</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) { 
      obj = obj || new exports();

      if (data.hasOwnProperty('name')) {
        obj['name'] = ApiClient.convertToType(data['name'], 'String');
      }
      if (data.hasOwnProperty('maxHitPoints')) {
        obj['maxHitPoints'] = ApiClient.convertToType(data['maxHitPoints'], 'Number');
      }
      if (data.hasOwnProperty('maxSpeed')) {
        obj['maxSpeed'] = ApiClient.convertToType(data['maxSpeed'], 'Number');
      }
      if (data.hasOwnProperty('acceleration')) {
        obj['acceleration'] = ApiClient.convertToType(data['acceleration'], 'Number');
      }
      if (data.hasOwnProperty('decelleration')) {
        obj['decelleration'] = ApiClient.convertToType(data['decelleration'], 'Number');
      }
      if (data.hasOwnProperty('maxSterlingSpeed')) {
        obj['maxSterlingSpeed'] = ApiClient.convertToType(data['maxSterlingSpeed'], 'Number');
      }
      if (data.hasOwnProperty('maxScanDistance')) {
        obj['maxScanDistance'] = ApiClient.convertToType(data['maxScanDistance'], 'Number');
      }
      if (data.hasOwnProperty('maxFireDistance')) {
        obj['maxFireDistance'] = ApiClient.convertToType(data['maxFireDistance'], 'Number');
      }
      if (data.hasOwnProperty('bulletSpeed')) {
        obj['bulletSpeed'] = ApiClient.convertToType(data['bulletSpeed'], 'Number');
      }
      if (data.hasOwnProperty('bulletDamage')) {
        obj['bulletDamage'] = ApiClient.convertToType(data['bulletDamage'], 'Number');
      }
      if (data.hasOwnProperty('cannonReloadingTime')) {
        obj['cannonReloadingTime'] = ApiClient.convertToType(data['cannonReloadingTime'], 'Number');
      }
    }
    return obj;
  }


  /**
   * @member {String} name
   */
  exports.prototype['name'] = undefined;

  /**
   * The initial health of the robot. Default value: 100.0
   * @member {Number} maxHitPoints
   */
  exports.prototype['maxHitPoints'] = undefined;

  /**
   * Max speed of the robot. Default value: 27.0 m/s
   * @member {Number} maxSpeed
   */
  exports.prototype['maxSpeed'] = undefined;

  /**
   * Acceleration of the robot. Default value 9.0 m/s^2
   * @member {Number} acceleration
   */
  exports.prototype['acceleration'] = undefined;

  /**
   * The max deceleration in case of reduction of the speed. A negative number. Default value -5.0 m/s^2
   * @member {Number} decelleration
   */
  exports.prototype['decelleration'] = undefined;

  /**
   * The maximim speed at which a robot can change direction.  Default value 13.0 m/s.\nIf the robot changes direction when it has a speed greater than this value, it start decelerating without changing direction, until it does not reach the steerling speed.\n
   * @member {Number} maxSterlingSpeed
   */
  exports.prototype['maxSterlingSpeed'] = undefined;

  /**
   * The max distance the robot can identify targets. Default value: 700 m/s.
   * @member {Number} maxScanDistance
   */
  exports.prototype['maxScanDistance'] = undefined;

  /**
   * The max distance a bullet can reach. Default value: 700 m/s.
   * @member {Number} maxFireDistance
   */
  exports.prototype['maxFireDistance'] = undefined;

  /**
   * The speed of a fired bullet. Default value: 500 m/s.
   * @member {Number} bulletSpeed
   */
  exports.prototype['bulletSpeed'] = undefined;

  /**
   * The max health damage a bullet can inflict when it reach exactly the target.\nDefault value: 10.0.\nActually max allowed value is 20 and minimum allowed value is 1.\nThe bullet hit exactly a target in case the explosion is within a certain distance, actually 2 meters.\nThe bullet can make a limited damage if it explodes near the target.\nThe limited damage decrease linearly from 50% to 0%, from 2 meters to 45 meters distance from the target.\nWhen you configure a robot, there is an associated a strenght using an heuristic, based on robot configured characteristics.\nThen the bulletDamage is increased until it does not reach the maximum possibile value.\n
   * @member {Number} bulletDamage
   */
  exports.prototype['bulletDamage'] = undefined;

  /**
   * How many seconds the robot must wait before firing another missile. Min value: 1s. Max value: 6s. Deafault value: 1s.\n
   * @member {Number} cannonReloadingTime
   */
  exports.prototype['cannonReloadingTime'] = undefined;




  return exports;
}));

File: client/generated_rest_api/java/src/main/java/io/swagger/client/model/RobotConfiguration.java
package io.swagger.client.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;



/**
 * Configurations of the robot.\nConfigurations can be sent during robot creation, or can be returned after robot creation.\n
 **/

@ApiModel(description = "Configurations of the robot.\nConfigurations can be sent during robot creation, or can be returned after robot creation.\n")
@javax.annotation.Generated(value = "class io.swagger.codegen.languages.JavaClientCodegen", date = "2016-04-30T12:28:53.526+02:00")
public class RobotConfiguration   {
  
  private String name = null;
  private Float maxHitPoints = null;
  private Float maxSpeed = null;
  private Float acceleration = null;
  private Float decelleration = null;
  private Float maxSterlingSpeed = null;
  private Float maxScanDistance = null;
  private Float maxFireDistance = null;
  private Float bulletSpeed = null;
  private Float bulletDamage = null;
  private Float fireReloadingTime = null;

  
  /**
   **/
  public RobotConfiguration name(String name) {
    this.name = name;
    return this;
  }
  
  @ApiModelProperty(example = "null", required = true, value = "")
  @JsonProperty("name")
  public String getName() {
    return name;
  }
  public void setName(String name) {
    this.name = name;
  }

  
  /**
   * The initial health of the robot. Default value: 100.0
   **/
  public RobotConfiguration maxHitPoints(Float maxHitPoints) {
    this.maxHitPoints = maxHitPoints;
    return this;
  }
  
  @ApiModelProperty(example = "null", value = "The initial health of the robot. Default value: 100.0")
  @JsonProperty("maxHitPoints")
  public Float getMaxHitPoints() {
    return maxHitPoints;
  }
  public void setMaxHitPoints(Float maxHitPoints) {
    this.maxHitPoints = maxHitPoints;
  }

  
  /**
   * Max speed of the robot. Default value: 27.0 m/s
   **/
  public RobotConfiguration maxSpeed(Float maxSpeed) {
    this.maxSpeed = maxSpeed;
    return this;
  }
  
  @ApiModelProperty(example = "null", value = "Max speed of the robot. Default value: 27.0 m/s")
  @JsonProperty("maxSpeed")
  public Float getMaxSpeed() {
    return maxSpeed;
  }
  public void setMaxSpeed(Float maxSpeed) {
    this.maxSpeed = maxSpeed;
  }

  
  /**
   * Acceleration of the robot. Default value 9.0 m/s^2
   **/
  public RobotConfiguration acceleration(Float acceleration) {
    this.acceleration = acceleration;
    return this;
  }
  
  @ApiModelProperty(example = "null", value = "Acceleration of the robot. Default value 9.0 m/s^2")
  @JsonProperty("acceleration")
  public Float getAcceleration() {
    return acceleration;
  }
  public void setAcceleration(Float acceleration) {
    this.acceleration = acceleration;
  }

  
  /**
   * The max deceleration in case of reduction of the speed. A negative number. Default value -5.0 m/s^2
   **/
  public RobotConfiguration decelleration(Float decelleration) {
    this.decelleration = decelleration;
    return this;
  }
  
  @ApiModelProperty(example = "null", value = "The max deceleration in case of reduction of the speed. A negative number. Default value -5.0 m/s^2")
  @JsonProperty("decelleration")
  public Float getDecelleration() {
    return decelleration;
  }
  public void setDecelleration(Float decelleration) {
    this.decelleration = decelleration;
  }

  
  /**
   * The maximim speed at which a robot can change direction.  Default value 13.0 m/s.\nIf the robot changes direction when it has a speed greater than this value, it start decelerating without changing direction, until it does not reach the steerling speed.\n
   **/
  public RobotConfiguration maxSterlingSpeed(Float maxSterlingSpeed) {
    this.maxSterlingSpeed = maxSterlingSpeed;
    return this;
  }
  
  @ApiModelProperty(example = "null", value = "The maximim speed at which a robot can change direction.  Default value 13.0 m/s.\nIf the robot changes direction when it has a speed greater than this value, it start decelerating without changing direction, until it does not reach the steerling speed.\n")
  @JsonProperty("maxSterlingSpeed")
  public Float getMaxSterlingSpeed() {
    return maxSterlingSpeed;
  }
  public void setMaxSterlingSpeed(Float maxSterlingSpeed) {
    this.maxSterlingSpeed = maxSterlingSpeed;
  }

  
  /**
   * The max distance the robot can identify targets. Default value: 700 m/s.
   **/
  public RobotConfiguration maxScanDistance(Float maxScanDistance) {
    this.maxScanDistance = maxScanDistance;
    return this;
  }
  
  @ApiModelProperty(example = "null", value = "The max distance the robot can identify targets. Default value: 700 m/s.")
  @JsonProperty("maxScanDistance")
  public Float getMaxScanDistance() {
    return maxScanDistance;
  }
  public void setMaxScanDistance(Float maxScanDistance) {
    this.maxScanDistance = maxScanDistance;
  }

  
  /**
   * The max distance a bullet can reach. Default value: 700 m/s.
   **/
  public RobotConfiguration maxFireDistance(Float maxFireDistance) {
    this.maxFireDistance = maxFireDistance;
    return this;
  }
  
  @ApiModelProperty(example = "null", value = "The max distance a bullet can reach. Default value: 700 m/s.")
  @JsonProperty("maxFireDistance")
  public Float getMaxFireDistance() {
    return maxFireDistance;
  }
  public void setMaxFireDistance(Float maxFireDistance) {
    this.maxFireDistance = maxFireDistance;
  }

  
  /**
   * The speed of a fired bullet. Default value: 500 m/s.
   **/
  public RobotConfiguration bulletSpeed(Float bulletSpeed) {
    this.bulletSpeed = bulletSpeed;
    return this;
  }
  
  @ApiModelProperty(example = "null", value = "The speed of a fired bullet. Default value: 500 m/s.")
  @JsonProperty("bulletSpeed")
  public Float getBulletSpeed() {
    return bulletSpeed;
  }
  public void setBulletSpeed(Float bulletSpeed) {
    this.bulletSpeed = bulletSpeed;
  }

  
  /**
   * The max health damage a bullet can inflict when it reach exactly the target.\nDefault value: 10.0.\nActually max allowed value is 20 and minimum allowed value is 1.\nThe bullet hit exactly a target in case the explosion is within a certain distance, actually 2 meters.\nThe bullet can make a limited damage if it explodes near the target.\nThe limited damage decrease linearly from 10% to 0%, from 2 meters to 45 meters distance from the target.\nWhen you configure a robot, it is associated a strenght using an heuristic, based on robot configured characteristics.\nThen the bulletDamage is increased until it does not reach the maximum value respecting the maximum allowed robot strenght.\n
   **/
  public RobotConfiguration bulletDamage(Float bulletDamage) {
    this.bulletDamage = bulletDamage;
    return this;
  }
  
  @ApiModelProperty(example = "null", value = "The max health damage a bullet can inflict when it reach exactly the target.\nDefault value: 10.0.\nActually max allowed value is 20 and minimum allowed value is 1.\nThe bullet hit exactly a target in case the explosion is within a certain distance, actually 2 meters.\nThe bullet can make a limited damage if it explodes near the target.\nThe limited damage decrease linearly from 10% to 0%, from 2 meters to 45 meters distance from the target.\nWhen you configure a robot, it is associated a strenght using an heuristic, based on robot configured characteristics.\nThen the bulletDamage is increased until it does not reach the maximum value respecting the maximum allowed robot strenght.\n")
  @JsonProperty("bulletDamage")
  public Float getBulletDamage() {
    return bulletDamage;
  }
  public void setBulletDamage(Float bulletDamage) {
    this.bulletDamage = bulletDamage;
  }

  
  /**
   * How many seconds the robot must wait before firing another missile. Deafault value: 1.0 s\nActually this value must be within 1.0 s and 6.0 s\n
   **/
  public RobotConfiguration fireReloadingTime(Float fireReloadingTime) {
    this.fireReloadingTime = fireReloadingTime;
    return this;
  }
  
  @ApiModelProperty(example = "null", value = "How many seconds the robot must wait before firing another missile. Deafault value: 1.0 s\nActually this value must be within 1.0 s and 6.0 s\n")
  @JsonProperty("fireReloadingTime")
  public Float getFireReloadingTime() {
    return fireReloadingTime;
  }
  public void setFireReloadingTime(Float fireReloadingTime) {
    this.fireReloadingTime = fireReloadingTime;
  }

  

  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    RobotConfiguration robotConfiguration = (RobotConfiguration) o;
    return Objects.equals(this.name, robotConfiguration.name) &&
        Objects.equals(this.maxHitPoints, robotConfiguration.maxHitPoints) &&
        Objects.equals(this.maxSpeed, robotConfiguration.maxSpeed) &&
        Objects.equals(this.acceleration, robotConfiguration.acceleration) &&
        Objects.equals(this.decelleration, robotConfiguration.decelleration) &&
        Objects.equals(this.maxSterlingSpeed, robotConfiguration.maxSterlingSpeed) &&
        Objects.equals(this.maxScanDistance, robotConfiguration.maxScanDistance) &&
        Objects.equals(this.maxFireDistance, robotConfiguration.maxFireDistance) &&
        Objects.equals(this.bulletSpeed, robotConfiguration.bulletSpeed) &&
        Objects.equals(this.bulletDamage, robotConfiguration.bulletDamage) &&
        Objects.equals(this.fireReloadingTime, robotConfiguration.fireReloadingTime);
  }

  @Override
  public int hashCode() {
    return Objects.hash(name, maxHitPoints, maxSpeed, acceleration, decelleration, maxSterlingSpeed, maxScanDistance, maxFireDistance, bulletSpeed, bulletDamage, fireReloadingTime);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class RobotConfiguration {\n");
    
    sb.append("    name: ").append(toIndentedString(name)).append("\n");
    sb.append("    maxHitPoints: ").append(toIndentedString(maxHitPoints)).append("\n");
    sb.append("    maxSpeed: ").append(toIndentedString(maxSpeed)).append("\n");
    sb.append("    acceleration: ").append(toIndentedString(acceleration)).append("\n");
    sb.append("    decelleration: ").append(toIndentedString(decelleration)).append("\n");
    sb.append("    maxSterlingSpeed: ").append(toIndentedString(maxSterlingSpeed)).append("\n");
    sb.append("    maxScanDistance: ").append(toIndentedString(maxScanDistance)).append("\n");
    sb.append("    maxFireDistance: ").append(toIndentedString(maxFireDistance)).append("\n");
    sb.append("    bulletSpeed: ").append(toIndentedString(bulletSpeed)).append("\n");
    sb.append("    bulletDamage: ").append(toIndentedString(bulletDamage)).append("\n");
    sb.append("    fireReloadingTime: ").append(toIndentedString(fireReloadingTime)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

